### DISCUSSION TITLE
How to drop an external item on ot the tree to create a new node

### DISCUSSION THREAD

**Discussion by martinmoesby** [nosferatu500/react-sortable-tree#4] [Q&A] [answered]
Is this a popssibility? I have a list of items, that I neede to be able to use as a source for creating new nodes in the tree, but for now I have only seen  the option from dragging from another tree.

**Comment by nosferatu500** [comment, marked as answer]
Hi @martinmoesby 

You can add a new node programmatically.

Take a look:
[add-and-remove-nodes-programmatically](https://github.com/nosferatu500/react-sortable-tree/blob/master/stories/add-remove.js)
[drag-from-external-source](https://github.com/nosferatu500/react-sortable-tree/blob/master/stories/external-node.js)

UPDATE:
[Live Demo](https://frontend-collective.github.io/react-sortable-tree/?path=/story/advanced--drag-from-external-source)
